deploy/crds/crd-certificates.yaml

```diff
@@ -1089,11 +1089,12 @@ spec:
                   type: object
                   properties:
                     algorithm:
-                      description: Algorithm is the private key algorithm of the corresponding private key for this certificate. If provided, allowed values are either `RSA` or `ECDSA` If `algorithm` is specified and `size` is not provided, key size of 256 will be used for `ECDSA` key algorithm and key size of 2048 will be used for `RSA` key algorithm.
+                      description: Algorithm is the private key algorithm of the corresponding private key for this certificate. If provided, allowed values are either `RSA`,`Ed25519` or `ECDSA` If `algorithm` is specified and `size` is not provided, key size of 256 will be used for `ECDSA` key algorithm and key size of 2048 will be used for `RSA` key algorithm. key size is ignored when using the `Ed25519` key algorithm.
                       type: string
                       enum:
                         - RSA
                         - ECDSA
+                        - Ed25519
```

Comment:
For consistency with other values for this field, can we make this all capitals? (ie ED25519)

Reply:
(Unless it makes more sense to have the 'd' lowercase? I see a lot of literature does refer to it in that way)

Reply:
I really prefer `Ed25519`, `ED25519` really jars for me being.

Notably, Go uses `Ed25519`:

```bash
$ go doc x509 PureEd25519
...
$ go doc x509 ed25519

const (
        ...
        Ed25519
)

```

Reply:
also the naming from go likely comes from this [RFC8410 ](https://datatracker.ietf.org/doc/html/rfc8410#section-8)where they talk about the naming consistency.